src/libs/ReportScrollManager/index.native.js

```diff
@@ -4,13 +4,33 @@ import React from 'react';
 // to the original ref.
 const flatListRef = React.createRef();
 
+// A reference to the last index required to scroll to, for async layout change handler.
+let lastIndex;
+
 /**
  * Scroll to the provided index.
  *
  * @param {Object} index
  */
 function scrollToIndex(index) {
     flatListRef.current.scrollToIndex(index);
+    lastIndex = index;
+}
+
+/**
+ * Scroll to the last saved index on layout change
+ * The is a fallback handler for asyncronius scrolling and layout changes.
+ */
+function scrollToLastIndex() {
+    if (!flatListRef.current) {
+        return;
+    }
+    if (lastIndex === undefined) {
+        return;
+    }
+
+    flatListRef.current.scrollToIndex(lastIndex);
+    lastIndex = undefined;
```

Comment:
Do we need to do this? I might be wrong, but I think we don't need to set lastIndex to `undefined` and don't need the `if (lastIndex === undefined)` check. Could you verify that, please?

Reply:
The uncontrolled behavior requires this because the events aren't chained and can happen asynchronously the check validates and clearance of this value validates the action won't happen again if not actually needed